src/frontend/js/components/SaleTunnel/index.tsx

```diff
@@ -105,6 +109,15 @@ const SaleTunnel = ({ product, courseRun, isOpen = false, onClose }: Props) => {
     reset();
   };
 
+  const context: SaleTunnelContextType = useMemo(
+    () => ({
+      product,
```

Comment:
We could provide the enrollment or the course implied in the purchase into this context. In this way, within the PaymentButton component, we could remove the use of `useCourseProduct` in favor of `useSaleTunnelContext`

Reply:
Ok I'm going to check how to achieve that :)